moonraker/components/wled.py

```diff
@@ -0,0 +1,445 @@
+                # All pixels same color only send range command
+                elem = []
+                for p in self._chain_data[0:elem_size]:
+                    elem.append(p)
+                self.send_full_chain_data = False
+                await self._send_wled_command(
+                    {"seg": {"i": [0, self.chain_count-1, elem]}})
+            elif self.send_full_chain_data:
+                # Send a full set of color data (e.g. previous preset)
+                state: Dict[str, Any] = {"seg": {"i": []}}
+                cdata = []
+                for i in range(self.chain_count):
+                    idx = i * elem_size
+                    elem = []
+                    for p in self._chain_data[idx: idx+elem_size]:
+                        elem.append(p)
+                    cdata.append(elem)
+                state["seg"]["i"] = cdata
+                self.send_full_chain_data = False
+                await self._send_wled_command(state)
+            else:
+                # Only one pixel has changed so send just that one
+                elem = []
+                for p in self._chain_data[(index-1)*elem_size:
+                                          (index-1)*elem_size+elem_size]:
+                    elem.append(p)
+                await self._send_wled_command({"seg": {"i": [index, elem]}})
+        else:
+            # If not transmitting this time easiest just to send all data when
+            # next transmitting
+            self.send_full_chain_data = True
+
+class WLED:
+    def __init__(self: WLED, config: ConfigHelper) -> None:
+        try:
+            # root_logger = logging.getLogger()
+            # root_logger.setLevel(logging.DEBUG)
+
+            self.server = config.get_server()
+            prefix_sections = config.get_prefix_sections("wled")
+            logging.info(f"WLED component loading strips: {prefix_sections}")
+            color_orders = {
+                "RGB": ColorOrder.RGB,
+                "RGBW": ColorOrder.RGBW
+            }
+            self.strips = {}
+            for section in prefix_sections:
+                cfg = config[section]
+
+                name_parts = cfg.get_name().split(maxsplit=1)
+                if len(name_parts) != 2:
+                    raise cfg.error(
+                        f"Invalid Section Name: {cfg.get_name()}")
+                name: str = name_parts[1]
+
+                logging.info(f"WLED strip: {name}")
+
+                color_order_cfg: str = cfg.get("color_order", "RGB")
+                color_order = color_orders.get(color_order_cfg)
+                if color_order is None:
+                    raise config.error(
+                        f"Color order not supported: {color_order_cfg}")
+
+                self.strips[name] = Strip(name, color_order, cfg)
+
+            event_loop = self.server.get_event_loop()
+            event_loop.register_callback(
+                self._initalize_strips, list(self.strips.values()))
```

Comment:
With the new changes to Moonraker initialization you no longer need this callback.  More details to follow in the next comment.

Reply:
Removed and moved